zero/src/provider.rs

```diff
@@ -118,3 +136,20 @@ where
         }
     }
 }
+
+impl<ProviderT, TransportT> ZeroBlockProvider for CachedProvider<ProviderT, TransportT>
```

Comment:
I don't think this is the right implementation hierarchy

Reply:
Why not?

Reply:
To add more context - `ZeroBlockProvider` is the API that `BlockInterval` needs to properly resolves block range that will be proven, and nothing more than that. If the name of the trait is too presumptuous, we should name it differently.

Reply:
Updated to `impl<T: Provider> BlockIntervalProvider for T {`